### PR TITLE
chore: align Expander with WinUI

### PR DESCRIPTION
**Pull request by ramezgerges**
GitHub Issue (If applicable): closes #

## PR Type

What kind of change does this PR introduce?
- Refactoring (no functional changes, no api changes)

### <samp>🤖 Generated by Copilot at eaba1b4</samp>

This pull request updates the Expander control and its tests to match the latest WinUI version. It modifies the `Expander.cs` and `ExpanderAutomationPeer.cs` files to improve the implementation and fix some issues. It also adds two new test files, `ExpanderTests_APITests.cs` and `ExpanderTests_InteractionTests.cs`, to cover the automation peer and interaction scenarios.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/Expander/Expander.cs [hunk 1]
```diff
@@ -75,6 +71,11 @@ protected override void OnApplyTemplate()
 			}
 		}
 
+		// Uno specific: Added to dispose event handlers
+		_eventSubscriptions.Disposable = null;
+		var registrations = new CompositeDisposable();
+		_eventSubscriptions.Disposable = registrations;
+
```
Comment (Youssef1313) [inline src/Uno.UI/Microsoft/UI/Xaml/Controls/Expander/Expander.cs:78]: I think we should keep the subscription and unsubscription in Loaded/Unloaded

Comment (ramezgerges) [reply src/Uno.UI/Microsoft/UI/Xaml/Controls/Expander/Expander.cs:78]: The pattern of subscribing to things in OnApplyTemplate occurs throughout the codebase (see e.g. Slider). This shouldn't be leaking as long as we're only subscribing to child elements. I confirmed this with @MartinZikmund.

Comment (MartinZikmund) [reply src/Uno.UI/Microsoft/UI/Xaml/Controls/Expander/Expander.cs:78]: Yes unless I am missing something, I think subscribing to a child should not leak 🤔 - do you see a problematic scenario there @Youssef1313 ?

Comment (Youssef1313) [reply src/Uno.UI/Microsoft/UI/Xaml/Controls/Expander/Expander.cs:78]: @MartinZikmund I'm not sure about iOS specifically where Swift GC can't handle circular references. https://learn.microsoft.com/en-us/xamarin/ios/deploy-test/performance#avoid-strong-circular-references I'm not sure if that's applicable here?

Comment (Youssef1313) [reply src/Uno.UI/Microsoft/UI/Xaml/Controls/Expander/Expander.cs:78]: I think one side of the circularity here is on the native side, and the other is managed, if I'm understanding correctly? Maybe it's not a problem in this case?

Comment (jeromelaban) [reply src/Uno.UI/Microsoft/UI/Xaml/Controls/Expander/Expander.cs:78]: Subscribing to a child will leak (through the delegate's target property) specifically on iOS. It's important to register/unregister to events in Load/Unload for that reason.

Comment (jeromelaban) [reply src/Uno.UI/Microsoft/UI/Xaml/Controls/Expander/Expander.cs:78]: We should also add that control the runtime leak tests (if not there already), to ensure we're not leaking:
https://github.com/unoplatform/uno/blob/93aa3e9825d6b3ee7d804649b40235569a3dd31a/src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml/Given_FrameworkElement_And_Leak.cs#L130

Comment (MartinZikmund) [reply src/Uno.UI/Microsoft/UI/Xaml/Controls/Expander/Expander.cs:78]: The odd thing is the control is already in the leak tests - so why does it not leak after this change according to CI 🤔 ?

Comment (MartinZikmund) [reply src/Uno.UI/Microsoft/UI/Xaml/Controls/Expander/Expander.cs:78]: Ah! Maybe it could be because the test is not using fluent V2 styles!

**Comment by MartinZikmund** [comment]
@ramezgerges ah sorry, I made this one https://github.com/unoplatform/uno/pull/14296 😓

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/Expander/Expander.cs (hunk 1 above)
Comment (MartinZikmund) [reply src/Uno.UI/Microsoft/UI/Xaml/Controls/Expander/Expander.cs:82]: The fluent leak tests uncovered two leaks on Android, so already paying off. The PR should get merged soon, so then we can rebase this one and see whether the leak occurs.

Comment (MartinZikmund) [reply src/Uno.UI/Microsoft/UI/Xaml/Controls/Expander/Expander.cs:82]: The leak tests were merged, can you rebase to see if it fails or not please?

**Pull request merged**